python/Generators/Generators.py

```diff
@@ -1,70 +1,35 @@
-                self.whizard = Whizard.Whizard(self.proc_info, self.settings)
-                self.whizard.write_file()
-                self.whizard.write_key4hepfile()
-            else:
-                print("Whizard module not found. Unable to initialize Whizard.")
-
-        if "Madgraph" in self.generator_list:
-            if Madgraph is not None:
-                self.madgraph = Madgraph.Madgraph(self.proc_info, self.settings)
-                self.madgraph.write_file()
-                self.madgraph.write_key4hepfile()
-            else:
-                print("Madgraph module not found. Unable to initialize Madgraph.")
-
-        if "Babayaga" in self.generator_list:
-            if Babayaga is not None:
-                self.babayaga = Babayaga.Babayaga(self.proc_info, self.settings)
-                self.babayaga.write_file()
-                self.babayaga.write_key4hepfile()
-            else:
-                print("Babayaga module not found. Unable to initialize Babayaga.")
-
-        if "KKMC" in self.generator_list:
-            if KKMC is not None:
-                self.kkmc = KKMC.KKMC(self.proc_info, self.settings)
-                self.kkmc.write_file()
-                self.kkmc.write_key4hepfile()
-            else:
-                print("KKMC module not found. Unable to initialize KKMC.")
 
-        if "Pythia" in self.generator_list:
-            if Pythia is not None:
-                self.pythia = Pythia.Pythia(self.proc_info, self.settings)
-                self.pythia.write_file()
-                self.pythia.write_key4hepfile()
-            else:
-                print("Pythia module not found. Unable to initialize Pythia.")
+        for generatorName in self.generator_list:
+            # get the module
+            try:
+                generator   = importlib.import_module(f"Generators.{generatorName}")
+                # get the ClassObject
+                generatorClass = getattr(generator,generatorName)
+                # execute the object
+                generatorObj = generatorClass(self.proc_info, self.settings)
+                #writing file
+                generatorObj.write_file()
+                #writing key4hep file 
+                generatorObj.write_key4hepfile()
+
+            except ModuleNotFoundError:
+                print(f"{generatorName} python module not found for {self.proc_info.get('_proclabel')}")
+            except AttributeError:
+                print(f"{generatorName} class could not be loaded with getattr for {self.proc_info.get('_proclabel')}")
+            except:
+                # all that remains is an excption from the execution of the modules
+                print(f"Execution of {generatorName} for {self.proc_info.get('_proclabel')} resulted in an exception, check the module for problems with loading doownstream modules like the corresponding ProcDB etc")
+                print("Datacard files and execution scripts not written for this generator")
```

Comment:
```suggestion
                print("Datacard files and execution scripts not written for this generator")
                raise
```
Re-raising the exception will give users more context regarding which exception was raised where. That is usually quite useful information.